packages/typescript-estree/src/convert.ts

```diff
@@ -417,2091 +470,2023 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
-        local: convertChild(node.name)
-      });
-      break;
+      // Modules
+      case SyntaxKind.ModuleBlock:
+        return this.createNode<es.TSModuleBlock>(node, {
+          type: AST_NODE_TYPES.TSModuleBlock,
+          body: this.convertBodyExpressions(node.statements, node)
+        });
 
-    case SyntaxKind.ImportSpecifier:
-      Object.assign(result, {
-        type: AST_NODE_TYPES.ImportSpecifier,
-        local: convertChild(node.name),
-        imported: convertChild(node.propertyName || node.name)
-      });
-      break;
+      case SyntaxKind.ImportDeclaration: {
+        const result = this.createNode<es.ImportDeclaration>(node, {
+          type: AST_NODE_TYPES.ImportDeclaration,
+          source: this.convertChild(node.moduleSpecifier),
+          specifiers: []
+        });
 
-    case SyntaxKind.ImportClause:
-      Object.assign(result, {
-        type: AST_NODE_TYPES.ImportDefaultSpecifier,
-        local: convertChild(node.name)
-      });
+        if (node.importClause) {
+          if (node.importClause.name) {
+            result.specifiers.push(this.convertChild(node.importClause));
+          }
 
-      // have to adjust location information due to tree differences
-      result.range[1] = node.name!.end;
-      result.loc = getLocFor(result.range[0], result.range[1], ast);
-      break;
+          if (node.importClause.namedBindings) {
+            switch (node.importClause.namedBindings.kind) {
+              case SyntaxKind.NamespaceImport:
+                result.specifiers.push(
+                  this.convertChild(node.importClause.namedBindings)
+                );
+                break;
+              case SyntaxKind.NamedImports:
+                result.specifiers = result.specifiers.concat(
+                  node.importClause.namedBindings.elements.map(el =>
+                    this.convertChild(el)
+                  )
+                );
+                break;
+            }
+          }
+        }
+        return result;
+      }
 
-    case SyntaxKind.ExportDeclaration:
-      if (node.exportClause) {
-        Object.assign(result, {
-          type: AST_NODE_TYPES.ExportNamedDeclaration,
-          source: convertChild(node.moduleSpecifier),
-          specifiers: node.exportClause.elements.map(convertChild),
-          declaration: null
-        });
-      } else {
-        Object.assign(result, {
-          type: AST_NODE_TYPES.ExportAllDeclaration,
-          source: convertChild(node.moduleSpecifier)
+      case SyntaxKind.NamespaceImport:
+        return this.createNode<es.ImportNamespaceSpecifier>(node, {
+          type: AST_NODE_TYPES.ImportNamespaceSpecifier,
+          local: this.convertChild(node.name)
         });
-      }
-      break;
 
-    case SyntaxKind.ExportSpecifier:
-      Object.assign(result, {
-        type: AST_NODE_TYPES.ExportSpecifier,
-        local: convertChild(node.propertyName || node.name),
-        exported: convertChild(n
```

Comment:
Again, `TODO` comment is unclear, please elaborate

Reply:
ohh, i missed this one, i will remove it

Reply:
while developing i'm adding a lot of todos to remind myself that i have to fix stuff